lib/stdlib/src/gen.erl

```diff
@@ -546,3 +548,41 @@ format_status_header(TagLine, RegName) when is_atom(RegName) ->
     lists:concat([TagLine, " ", RegName]);
 format_status_header(TagLine, Name) ->
     {TagLine, Name}.
+
+-spec format_status(Mod :: module(), Opt :: terminate | normal, Status, Args) ->
+          ReturnStatus when
+      Status :: #{ atom() => term() },
+      ReturnStatus :: #{ atom() => term(), '$status' => term()},
+      Args :: list(term()) | undefined.
+format_status(Mod, Opt, Status, Args) ->
+    case {erlang:function_exported(Mod, format_status, 1),
+          erlang:function_exported(Mod, format_status, 2)} of
+        {true, _} ->
+            try Mod:format_status(Status) of
```

Comment:
The PDict is not passed to format_status. Which makes following [code pattern](https://github.com/apache/couchdb/blob/0059b8f90e58e10b199a4b768a06a762d12a30d3/src/couch/src/couch_file.erl#L552:L554) impossible.

```
format_status(_Opt, [PDict, #file{} = File]) ->
    {_Fd, FilePath} = couch_util:get_value(couch_file_fd, PDict),
    [{data, [{"State", File}, {"InitialFilePath", FilePath}]}].
```

Reply:
If you use this code:

```erlang
format_status(_Opt, [PDict, #file{} = File]) ->
    {_Fd, FilePath} = couch_util:get_value(couch_file_fd, get()),
    [{data, [{"State", File}, {"InitialFilePath", FilePath}]}].
```

you will get the result that you are after. Or am I missing some detail where that is not possible?

Reply:
Yes this would work, but according to the modified docs in this PR it would mean relying on [deprecated](https://github.com/erlang/otp/pull/4952/files#diff-72eeebfa525d4fd117bb50d045278a11e3695660abf9bc5cbe38581690241de3R920) functionality.

Reply:
Actually I didn't notice a slight change you did. Let me review again.